spec/unit/outputs/kafka_spec.rb

```diff
@@ -131,7 +134,7 @@
 
       it "should retry until successful" do
         count = 0
-
+        success = nil
         expect_any_instance_of(org.apache.kafka.clients.producer.KafkaProducer).to receive(:send)
               .exactly(sendcount).times
               .and_wrap_original do |m, *args|
```

Comment:
I _believe_ your changes in the block make it so we no longer need to wrap the original, since `m` is no longer touched.

```suggestion
              .exactly(sendcount).times do |*args|
```